pkg/applicationmanager/controllers/backupsync.go

```diff
@@ -0,0 +1,156 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/libopenstorage/stork/pkg/crypto"
+	"github.com/libopenstorage/stork/pkg/log"
+	"github.com/libopenstorage/stork/pkg/objectstore"
+	"github.com/portworx/sched-ops/k8s"
+	"github.com/sirupsen/logrus"
+	"gocloud.dev/blob"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/record"
+)
+
+// BackupSyncController reconciles applicationbackup objects
+type BackupSyncController struct {
+	Recorder     record.EventRecorder
+	SyncInterval time.Duration
+	stopChannel  chan os.Signal
+}
+
+// Init Initializes the backup sync controller
+func (b *BackupSyncController) Init(stopChannel chan os.Signal) error {
+	b.stopChannel = stopChannel
+	go b.startBackupSync()
+	return nil
+}
+
+func (b *BackupSyncController) startBackupSync() {
+	for {
+		select {
+		case <-time.After(b.SyncInterval):
+			backupLocations, err := k8s.Instance().ListBackupLocations("")
+			if err != nil {
+				logrus.Errorf("Error getting backup location to sync: %v", err)
+				continue
+			}
+			for _, backupLocation := range backupLocations.Items {
+				err := b.syncBackupsFromLocation(&backupLocation)
+				if err != nil {
+					log.BackupLocationLog(&backupLocation).Errorf("Error syncing backups from location: %v", err)
+					continue
+				}
+			}
+
+		case <-b.stopChannel:
+			return
+		}
+	}
+}
+
+func (b *BackupSyncController) syncBackupsFromLocation(location *storkv1.BackupLocation) error {
+	if !location.Location.Sync {
+		return nil
+	}
+	bucket, err := objectstore.GetBucket(location)
+	if err != nil {
+		return err
+	}
+	iterator := bucket.List(&blob.ListOptions{
+		Prefix:    location.Namespace + "/",
+		Delimiter: "/",
+	})
+	backups := make(map[string]bool)
+	for {
+		object, err := iterator.Next(context.TODO())
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if object.IsDir {
+			backups[object.Key] = true
+		}
+	}
+	logrus.Infof("Backups in namespace %v: %v", location.Namespace, backups)
+	for backupName := range backups {
+		iterator := bucket.List(&blob.ListOptions{
+			Prefix:    backupName,
+			Delimiter: "/",
+		})
+		for {
+			object, err := iterator.Next(context.TODO())
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			if object.IsDir {
+				data, err := bucket.ReadAll(context.TODO(), filepath.Join(object.Key, metadataObjectName))
+				if err != nil {
+					continue
```

Comment:
we may want to log errors before `continue` here

Reply:
Done